platform/api/esx/api.go

```diff
@@ -585,7 +637,7 @@ func (a *API) updateOVFEnv(vm *object.VirtualMachine, userdata string) error {
 	}
 
 	for _, item := range mvm.Config.VAppConfig.(*types.VmConfigInfo).Property {
-		if item.Id == "guestinfo.coreos.config.data" {
+		if item.Id == "guestinfo.ignition.config.data" {
```

Comment:
This change (and the next one) seems unrelated to the overall change and I see no mention of it in the commit.

Was this a bug that wasn't working before? Could this affect something that was already configured with the wrong keys?

Reply:
I've added it to the commit message. The guest variables are set directly in one case and I used the new generic names there, and also updated them here.